### ISSUE TITLE
bug: Bad Local Variable Type

### ISSUE THREAD

**Issue by Ultimushadow** [adibarra/enchant-tweaker#4]
### What happened?

Attempted to launch the game with the most recent update of this mod (along side.. hundreds of others) and the game will not launch.

### What should have happened?

Game should have, you know, loaded up lol

### How to reproduce?

I have too many mods to list them here but simply download everything I have installed I suppose

### What is the jar file name?

enchanttweaker-1.4.0+mc1.19.4

### Any relevant logs?

[latest.log](https://github.com/adibarra/enchant-tweaker/files/11546143/latest.log)

### Anything else?

I have no idea where to begin with this error code so I'm hoping you do. Pretty much every reference to this issue online is some forge/optifine conflict which, obviously, is not the case here. I looked into the code itself to see if I could tell what was going on but the crash is helpfully not very specific on details to my untrained eyes so I couldn't get far. I have loads of mods installed, and I had just recently removed some mods that overlapped features as this was my intended replacement. If you need any information I will provide it as long as I know what I actually need to look for and send in.

### Before submitting...

- [X] This issue is not a duplicate of an existing issue (open or closed).

**Comment by GiienaAstrella** [comment]
It's worth noting that this issue applies to the client.

From the pack, it looks like [Memory Leak Fix](https://modrinth.com/mod/memoryleakfix) is the mod conflicting with Enchant Tweaker. I've attached [latest.log](https://mclo.gs/KOkkzwP). Versions are listed below.

**Enchant Tweaker Version:** 1.4.0+mc1.19.4
**Memory Leak Fix Version:** 1.0.0
**Fabric Loader Version:** 0.14.19
**Fabric API Version:** 0.81.1
**Minecraft Version:** 1.19.4
**Java Version:**
```
java version "17.0.2" 2022-01-18 LTS
Java(TM) SE Runtime Environment (build 17.0.2+8-LTS-86)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.2+8-LTS-86, mixed mode, sharing)
```

**Comment by adibarra** [comment]
After a little bit of investigation it seems like this is [a pre-existing bug](https://togithub.com/fxmorin/MemoryLeakFix/issues/95) in Memory Leak Fix. It does however seem to be triggered by EnchantTweaker being present. Still looking into it, but it seems to be caused by a shared dependency.

While I look for a solution, all I can suggest is using either MemoryLeakFix or EnchantTweaker. (I would recommend MemoryLeakFix)

**Comment by adibarra** [comment]
Looks like it [was an issue](https://togithub.com/chronosacaria/MCDungeonsWeapons/pull/216#issuecomment-1551934048) with the shared dependency and they released an update that fixes the issue. Will be included in the next update (v1.4.2). Thanks for the bug report!

**Comment by Ultimushadow** [comment]
Looks like you have the solution down, but I did want to add some extra findings in case it helps
Client with MemoryLeakFix will also crash when joining a server with EnchantTweaker specifically if the config for free prior work is enabled, works fine without that config enabled though, I'm sure given you've already checked the dependency issue that you know why this is and it'll be patched out soon.
